### ISSUE TITLE
WebGL Context Lost in Multi-tab Dashboard for Deck.gl Scatter Plot

### ISSUE THREAD

**Issue by santosh-sahoo334** [apache/superset#24016]
### Discussed in https://github.com/apache/superset/discussions/23928

<div type='discussions-op-text'>

<sup>Originally posted by **santosh-sahoo334** May  4, 2023</sup>
Hi Team,
Not sure if this question was asked before, but facing some weird behavior with deck.gl scatter plot charts. I have a dashboard with 10 Tabs and 40 Charts. Each Tab with 4 charts - all are of deck.gl scatter plot type. When the dashboard loads for the first time, all charts rendered correctly, but after sometime ( like 5 mins ), when I switch between the tabs, I see the chart disappears with a small image icon at the to left as shown below :-
When I refresh the page again, charts appears to be normal. 
![image](https://user-images.githubusercontent.com/116371351/236133284-cd5edd09-a76b-4f78-a5fb-1857807c0624.png)

On inspecting found that -  too many WebGL contexts being loaded and finally lost. 
![image (1)](https://user-images.githubusercontent.com/116371351/236133495-b64bdeb5-0baf-4aed-8724-4068c1e44d0a.png)

Tried creating a Dashboard without any tab, with 7 charts (all of deck.gl scatter plot ), the same issue. Has anyone faced similar issue ?
Thanks!</div>

**Comment by rusackas** [comment]
There's a thread on [Superset Slack](http://bit.ly/join-superset-slack) about this here:
https://apache-superset.slack.com/archives/C015WAZL0KH/p1682058583475049

TL;DR here is that the browser (at least Chrome) allows no more than 6 web GL contexts (and one might need to configure it to allow > 4). 

Maybe @kgabryje knows this, but I'm wondering if components on hidden tabs are de-rendered. If so, you _should_ be able to have 4 DeckGL charts per tab, and as many tabs as you want. If that's NOT the case, we should probably find a way to do so, perhaps as part of the `DASHBOARD_VIRTUALIZATION` feature flag. 

Other solutions to this to allow unlimited WebGL contexts on a page involve some complex re-architecture, or some really gross hackery. We might need to impose a limitation/warning about this so dashboard authors can't get themselves int this situation.

**Comment by rusackas** [comment]
I think this is fixed now, and is pretty out of date if not. If people are still encountering this in current versions (3.x) please open a new Issue or a PR to address the problem.

**Comment by greg-om** [comment]
> I think this is fixed now, and is pretty out of date if not. If people are still encountering this in current versions (3.x) please open a new Issue or a PR to address the problem.

I just wanted to post this here for anyone else who stumbles across this thread.  This is NOT an issue with superset at all and is very specific to browsers themselves.  It looks like the amount of webgl contexts is a moving target and is very dependent on what OS browser etc. If you want to test what yours is, try out this tool: https://max-gl-contexts.glitch.me/ (I'm not sure who made it otherwise I'd give kudos!

Upgrading / altering superset won't help this issue!  For me, using Firefox allowed for 200 webgl contexts as opposed to Chrome's 8...  Just one more reason to use Firefox 😁. I'm on Ubunut 24.04.
